modules/shared/src/main/scala/org/tessellation/schema/BlockAsActiveTip.scala

```diff
@@ -1,10 +1,27 @@
 package org.tessellation.schema
 
+import cats.Order
+
+import org.tessellation.ext.cats.data.OrderBasedOrdering
+import org.tessellation.schema.block.DAGBlock
 import org.tessellation.security.signature.Signed
 
+import derevo.cats.show
+import derevo.circe.magnolia.{decoder, encoder}
+import derevo.derive
 import eu.timepit.refined.types.numeric.NonNegLong
 
-trait BlockAsActiveTip[B <: Block[_]] {
-  val block: Signed[B]
-  val usageCount: NonNegLong
+@derive(show, encoder, decoder)
+case class BlockAsActiveTip[B <: Block[_]](
+  block: Signed[B],
+  usageCount: NonNegLong
+)
+
+object BlockAsActiveTip {
+  implicit def order[B <: Block[_]: Order]: Order[BlockAsActiveTip[B]] = Order.whenEqual(
+    Order.by[BlockAsActiveTip[B], Signed[B]](_.block),
+    Order.by[BlockAsActiveTip[B], NonNegLong](_.usageCount)
+  )
+
+  implicit object OrderingInstance extends OrderBasedOrdering[BlockAsActiveTip[DAGBlock]]
```

Comment:
hm, looks like mixed abstraction with implementation (`DAGBlock` is a specific implementation)

Reply:
I do not like it either. I think we can just keep it next to `DAGBlock`.